float.js

```diff
@@ -1,327 +1,207 @@
-var ajax = new XMLHttpRequest();
-var currentid = "";
-var currenturl = "";
-
-ajax.onreadystatechange = function() {
-    if (ajax.readyState == 4) {
-        if (ajax.status == 200) {
-            // We got item info, parse it and send it to the DOM
-            var json = JSON.parse(ajax.responseText);
-            if ("iteminfo" in json) {
-                window.postMessage({source: "contentscript", type: "success", iteminfo: json["iteminfo"],
-                    id: currentid, host: "CSGOFloat"}, "*");
-            }
-        }
-        else {
-            // Send the error to the DOM
-            window.postMessage({source: "contentscript", type: "error", error: JSON.parse(ajax.responseText),
-                id: currentid, host: "CSGOFloat"}, "*");
-        }
+let floatQueue = [];
+let floatData = {};
+let floatTimer;
+let steamListingInfo = {};
+
+// retrieve g_rgListingInfo from page script
+window.addEventListener('message', (e) => {
+    steamListingInfo = e.data.listingInfo;
+});
+
+let retrieveListingInfoFromPage = function(listingId) {
+    if (listingId != null && (listingId in steamListingInfo)) {
+        return Promise.resolve(steamListingInfo[listingId]);
     }
+
+    let script = document.createElement('script');
+    script.innerText = `
+        window.postMessage({
+            listingInfo: g_rgListingInfo
+        }, '*');
+    `;
+    document.head.appendChild(script);
```

Comment:
I just realized we could probably use a `postMessage` from the extension -> the page and only add a single script tag like you do in the current version.